Add `--config` param for flake8

mpanarin opened the pull request:
Allow explicit pass of config to flake. This can be useful for weird per-project configs that are not in default directories.
 
@gatesn is this config (https://github.com/palantir/python-language-server/blob/develop/pyls/config/flake8_conf.py#L32) used somewhere?
I was unable to find where :\

mpanarin committed:
[add] `--config` param for flake8

ccordoba12 commented:
@youben11, could you take a look at this one? Thanks!

youben11 commented:
@mpanarin flake8_conf.py isn't used by default, but if it's mentioned [here](https://github.com/palantir/python-language-server/blob/develop/pyls/config/config.py#L16) then it would load flake8 conf file on startup. Could also be configured via vscode client.

I think this PR is about specifying a configuration file to use after loading the project configuration file (.flake8, setup.cfg, tox.ini), am I right? Because the 'config' option here would be parsed from one of the project config file to be loaded as 'plugins.flake8.config'

mpanarin commented:
@youben11 yes, I saw that it is imported there. I just don't understand how it is used further down with flake8 tool. Especially as `flake8` is now called with Popen, instead of an api. Are you sure that is not a dead code?

Anyway, this pr is about specifying the config file to the cli tool, yes. 
Its WIP, I need to check a few thing

youben11 commented:
@mpanarin flake8_config will load config as 'plugins.flake8.*' then the flake8 plugin will use them for linting, you can see [here](https://github.com/palantir/python-language-server/blob/develop/pyls/plugins/flake8_lint.py#L18) how config values are used, this function will be called each time a lint operation is to be done (if the plugin is enabled).

I may be wrong about the usecase, but I wonder if we should parse that option from pyls settings (that can be configured by clients) instead of parsing that from a config file?

mpanarin commented:
> I may be wrong about the usecase, but I wonder if we should parse that option from pyls settings (that can be configured by clients) instead of parsing that from a config file?

That is exactly what I intended to do. I updated the PR and removed the param from flake8_config. Checked for it to work in emacs environment. 

Should I add tests for this pretty small change?
Also, I am not sure where to note, that other parameters such as `max-line-length` overwrite params from config. Should it be noted on the specific frontend insteaad? (VScode/emacs/etc.)

youben11 commented:
I think a quick test for ensuring the config option is working as intended would be great.

@ccordoba12 I'm note sure about parameters documentation.

mpanarin commented:
@youben11 @ccordoba12 updated

youben11 commented on pyls/plugins/flake8_lint.py:
```diff
@@ -64,13 +74,15 @@ def build_args(options, doc_path):
-        elif isinstance(arg_val, int):
+        else:
             arg = '--{}={}'.format(arg_name, arg_val)
         if arg:
```
I think we should get rid of this check as you made sure `arg` always holds a value

mpanarin replied:
updated

mpanarin commented:
@ccordoba12 Hello,
PR is ready and reviews are attended, can this be merged, please?)

ccordoba12 committed:
Merge branch 'develop' into config_param_for_flake

ccordoba12 commented:
Sorry, since I saw it marked as WIP I thought it was not ready.
